src/dsp_tools/utils/xml_upload.py

```diff
@@ -171,30 +172,37 @@ def _remove_circular_references(
 
     # sort the resources according to outgoing resptrs
     ok_resources: list[XMLResource] = []
+    # resources with circular references
     nok_resources: list[XMLResource] = []
+    # internal ids for the resources that do not have circular references
     ok_res_ids: list[str] = []
     cnt = 0
     nok_len = 9999999
     while len(resources) > 0 and cnt < 10000:
         for resource in resources:
             resptrs = resource.get_resptrs()
+            # get all the resptrs which have an internal id, i.e. that do not exist in the triplestore
             resptrs = [x for x in resptrs if not regex.search(r"https?://rdfh.ch/[a-fA-F0-9]{4}/\w{22}", x)]
+            # if there are no resptrs references, append to the ok resources
             if len(resptrs) == 0:
                 ok_resources.append(resource)
                 ok_res_ids.append(resource.id)
             else:
                 ok = True
+                # iterate over the list with all the resptrs that have internal links
                 for resptr in resptrs:
+                    # if that resptr is not in the ok list, set the flag to false
                     if resptr not in ok_res_ids:
                         ok = False
+                # if all the resptr are in the ok list, then there are no circular references
                 if ok:
                     ok_resources.append(resource)
                     ok_res_ids.append(resource.id)
+                # if any of the resptr are not in the ok list append the resource to the not ok list
                 else:
                     nok_resources.append(resource)
         resources = nok_resources
         if len(nok_resources) == nok_len:
-            # there are circular references. go through all problematic resources, and stash the problematic references.
```

Comment:
This comment has important info in it. The check `if len(nok_resources) == nok_len` is not clear enough, so I need the info `# there are circular references`

The rest of the comment can be removed ("go through...")